### PR TITLE
[OSD-7582] Refactor makefile  and dev improvement 

### PR DESCRIPTION
**Pull request by mrWinston**
This PR extracts the makefile changes from [this](https://github.com/openshift/aws-account-shredder/pull/62) PR into its own.
A couple of highlights include:
* Validation of correct target kubernetes cluster ( so we don't accidentally remove stuff in prod clusters )
* Wrap development prerequisites in make targets
* Improve documentation around local development
* Use prod deployment templates in local deployments ( removed existing yamls for local development )

## quick test

To test the changes, follow the instruction in the newly updated README.md.

jira: [OSD-7582](https://issues.redhat.com/browse/OSD-7582)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,32 @@ GO=CGO_ENABLED=0 GOOS=$(GOOS) GOARCH=$(GOARCH) GO111MODULE=on go
...
+.PHONY: check-aws-account-credentials
+check-aws-account-credentials: ## Check if AWS Account Env vars are set
+ifndef AWS_ACCESS_KEY_ID
+	$(error AWS_ACCESS_KEY_ID is undefined)
+endif
+ifndef AWS_SECRET_ACCESS_KEY
+	$(error AWS_SECRET_ACCESS_KEY is undefined)
+endif
+
+only-local-ctx:
```
Comment (dkeohane) [inline Makefile:43]: I would leave a small comment above this indicating that we try enforce to only run-local

File: README.md [hunk 2]
```diff
@@ -6,89 +6,69 @@
...
-
-```bash
-osdctl account set aws-shredder-account-delete --state=Failed
+make shred-account
 ```
 
 Once you set the status of the account to failed, the Shredder should pick it up and start shredding through the accounts.
 
-You should be able to follow the logs and watch the shred happen using `oc logs -f [pod name] -n aws-account-shredder`.  Certain objects may not delete on the first attempt through the shredder, but the shredder will continue to run on the account until it is created.
+You should be able to follow the logs and watch the shred happen using `make get-logs`.  Certain objects may not delete on the first attempt through the shredder, but the shredder will continue to run on the account until it is created.
```
Comment (dkeohane) [inline README.md:67]:
```suggestion
You should be able to follow the logs and watch the shred happen using `make get-logs`.  Certain aws resources may not completely delete on the first attempt through the shredder, but the shredder will continue to run on the account until it is cleaned.
```

File: README.md [hunk 3]
```diff
@@ -108,8 +88,5 @@ This will build the container image and spit out a tag at the end, now we'll nee
 ```
 docker push quay.io/<username>/aws-account-shredder:<tag>
 ```
-You'll now need to update `deploy/deployment.yaml` to point at the image you just built:
-```
-image: quay.io/<username>/aws-account-shredder:<image tag>
-```
-Now that you have the your image pushed up and your deployment updated, you can simply apply the deployment to roll out your local test branch.
+
+Now that you have the your image pushed up, you can simply apply the deployment again using `make deploy`
```
Comment (dkeohane) [inline README.md:92]:
```suggestion
Now that you have your image pushed up, you can simply apply the deployment again using `make deploy`
```

**Review by dkeohane — APPROVED** [review]
Awesome changes, love to see this! :t-rex: 

Only thing I'm unsure of here is the removal of the CRD yamls in the deploy folder, not sure what could be using them

File: Makefile (hunk 1 above)
Comment (mrWinston) [reply Makefile:44]: Done

File: README.md [hunk 4]
```diff
@@ -108,8 +88,5 @@ This will build the container image and spit out a tag at the end, now we'll nee
 ```
 docker push quay.io/<username>/aws-account-shredder:<tag>
 ```
-You'll now need to update `deploy/deployment.yaml` to point at the image you just built:
-```
-image: quay.io/<username>/aws-account-shredder:<image tag>
-```
-Now that you have the your image pushed up and your deployment updated, you can simply apply the deployment to roll out your local test branch.
+
+Now that you have your image pushed up, you can simply apply the deployment again using `make deploy`
```
Comment (NautiluX) [inline README.md:92]: does the user need to set a IMG env variable or something like that so the make will use the correct image?

Comment (NautiluX) [reply README.md:92]:
```suggestion
Now that you have your image pushed up, you can apply the deployment again using `make deploy`
```
words like simply or easy should be banned from all documentation. What's simple for you, can be complicated for someone else ;)

Comment (mrWinston) [reply README.md:92]: > does the user need to set a IMG env variable or something like that

Setting the image repository is documented a couple of lines above the one i've changed, so it did not appear in the diff:
```
You'll need to edit the `IMAGE_REPOSITORY` field in `project.mk` to point at your `aws-account-shredder` repo 
```

Comment (mrWinston) [reply README.md:92]: Alternatively, ( and probably better ), one can override the variable on the commandline or with an env var:
```
make docker-build IMAGE_REPOSITORY=maschulz
```
and
```
export IMAGE_REPOSITORY=maschulz
make docker-build
```
achieve exactly the same. Do you have a preference for what should be documented as the "default" way?

**Review by NautiluX — COMMENTED** [review]
/lgtm

File: README.md (hunk 4 above)
Comment (mrWinston) [reply README.md]: I added a line to the readme to clarify this a bit.

**Review by NautiluX — COMMENTED** [review]
/lgtm

**Pull request merged**